### PR TITLE
*: reset metrics after the leader steps down

### LINKED ISSUE #1415: Alert never stops after PD transfer leader
Currently, if the PD leader is under some kinds of unhealthy state, it might trigger the alarm depending on our alert rules. At that time, if the original leader transfers leader to another PD. The alert will never stop until we restart it.

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Closes #1415.

### What is changed and how it works?
This PR resets the metrics when the leader steps down.

### Check List 

Tests 

 - Manual test (by transferring the PD leader manually)

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -1014,6 +1016,19 @@ func (c *RaftCluster) collectMetrics() {
 	c.collectHealthStatus()
 }
 
+func (c *RaftCluster) resetMetrics() {
+	statsMap := statistics.NewStoreStatisticsMap(c.opt, c.GetNamespaceClassifier())
+	stores := c.GetStores()
+	for _, s := range stores {
+		statsMap.ResetStore(s, c.storesStats)
+	}
+	statsMap.Reset()
```
[the remark this answers is not in the document]
Comment (rleungx) [reply server/cluster.go]: Here we just use `statsMap` to reset the metrics. After `ResetStore` is called, the `statsMap` will not be an empty object.

File: server/coordinator.go [hunk 2]
```diff
@@ -368,7 +376,27 @@ func (c *coordinator) collectHotSpotMetrics() {
...
+	c.RLock()
+	defer c.RUnlock()
+	// Resets hot write region metrics.
+	stores := c.cluster.GetStores()
+	for _, s := range stores {
+		storeAddress := s.GetAddress()
+		storeID := s.GetID()
+		storeLabel := fmt.Sprintf("%d", storeID)
+		// Resets hot write metrics
+		hotSpotStatusGauge.WithLabelValues(storeAddress, storeLabel, "total_written_bytes_as_peer").Set(0)
```
Comment (nolouch) [inline server/coordinator.go]: Can we get all the `label values`? It's easy to ignore some clean up when we add new metrics. It seems the` Desc()` can get the `LabelPair`.

File: server/statistics/region_collection.go [hunk 3]
```diff
@@ -159,6 +159,11 @@ func (r *RegionStatistics) Collect() {
 	regionStatusGauge.WithLabelValues("empty-region-count").Set(float64(len(r.stats[EmptyRegion])))
 }
 
+// Reset resets the metrics of the regions' status.
```
Comment (lhy1024) [inline server/statistics/region_collection.go]:
```suggestion
// Reset resets the metrics of the region status.
```

File: server/statistics/store_collection.go [hunk 4]
```diff
@@ -177,15 +177,19 @@ func (s *storeStatistics) Collect() {
...
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "region_count")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "leader_size")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "leader_count")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "store_available")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "store_used")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "store_capacity")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "store_write_rate_bytes")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "store_read_rate_bytes")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "store_write_rate_keys")
+	storeStatusGauge.DeleteLabelValues(s.namespace, storeAddress, id, "store_read_rate_keys")
```
Comment (lhy1024) [inline server/statistics/store_collection.go]: It seemed to be more simple to use `for`?

**Comment by rleungx** [comment]
PTAL @shafreeck @lhy1024

**Comment by nolouch** [comment]
PTAL @shafreeck

**Pull request merged**